### PR TITLE
Modified robot docking state transitions

### PR DESCRIPTION
**Pull request by charizarrd**
- New docking state for faster scanning
- Change robot state to lost if it doesn't see anything while scanning
- Adjust angle in find_stream
- Move backwards for longer when bumped

### CODE REVIEW

File: kobuki_dock_drive/src/dock_drive_states.cpp [hunk 1]
```diff
@@ -157,8 +210,12 @@ namespace kobuki {
...
       if(left & (DockStationIRState::FAR_RIGHT + DockStationIRState::NEAR_RIGHT)) {
         next_state = RobotDockingState::GET_STREAM;
-        next_vx = 0.5;
-        next_wz = 0.0;
+        next_vx = 0.05;
+
+      	// slight hack: smaller dock_detector values indicate robot is farther away from dock so
+      	// needs greater angle correction, should probably test around some more to find what
+        // function would work best for this
+        next_wz = 8.0*(1.0 - (double)dock_detector/160.0);
```
Comment (KaijenHsiao) [inline kobuki_dock_drive/src/dock_drive_states.cpp]: what's with the 8*(1-dock_detector/160) business?  What values does dock_detector take?

Comment (charizarrd) [inline kobuki_dock_drive/src/dock_drive_states.cpp]: Oops, that was my hack for fixing when the robot starts out to the very far left/right of the dock and it tries to move towards the center of the dock, it starts moving at an angle away from the dock and ends up getting lost most of the time. This only really happens if it is pretty far away from the dock though so we could probably get rid of it; it's definitely not the best solution.

dock_detector is a variable that sort of indicates how far away the robot is from the dock: https://github.com/mayfieldrobotics/kobuki_core/blob/kobuki_docking/kobuki_dock_drive/src/dock_drive_states.cpp#L100. Higher values indicate the robot is closer.

**Comment by charizarrd** [comment]
@KaijenHsiao added longer explanation! Maybe too wordy? XD

**Comment by KaijenHsiao** [comment]
Nope, that's great. :)  I'll merge the whole lot now.
